### PR TITLE
Update the readme with more information about thread safety and other…

### PR DESCRIPTION
**Pull request by AlexGhiondea**
… concepts.

/cc @heaths @pakrym @maririos

### CODE REVIEW

**Comment by pakrym** [comment]
Why only these libraries were selected?

**Comment by AlexGhiondea** [comment]
@pakrym I wanted to start with 1 library and make sure the format is good before fixing the rest.

**Review by pakrym — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Administration/README.md [hunk 1]
```diff
@@ -148,6 +148,20 @@ A `BackupOperation` represents a long running operation for a full key backup.
...
 A `RestoreOperation` represents a long running operation for both a full key and selective key restore.
 
+### Thread safety
+We guarantee that all client instance methods are thread-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-methods-thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across threads.
+
+### Additional concepts
+<!-- CLIENT COMMON BAR -->
+[Client options](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/README.md#configuring-service-clients-using-clientoptions) | 
+[Accessing Response](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/README.md#accessing-http-response-details-using-responset) |
+[Long-running operations](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/README.md#accessing-http-response-details-using-responset) |
```
Comment (maririos) [inline sdk/keyvault/Azure.Security.KeyVault.Administration/README.md:158]:
```suggestion
[Long-running operations](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/README.md#consuming-long-running-operations-using-operationt) |
```

File: sdk/keyvault/Azure.Security.KeyVault.Administration/README.md [hunk 2]
```diff
@@ -148,6 +148,20 @@ A `BackupOperation` represents a long running operation for a full key backup.
...
+[Handling failures](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/README.md#reporting-errors-requestfailedexception) |
+[Logging](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/README.md#setting-up-console-logging) |
```
Comment (maririos) [inline sdk/keyvault/Azure.Security.KeyVault.Administration/README.md:160]: should this better point to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Diagnostics.md ?

Comment (maririos) [inline sdk/keyvault/Azure.Security.KeyVault.Administration/README.md:160]: Should this be moved/replicated in the `Troubleshooting` section?

**Review by heaths — APPROVED** [review]
Why only Key Vault? I don't see any sections about thread safety in Core, which seems like a good place to start (since it's really the one initially guaranteeing thread safety) if we're setting a new README template standard.

File: sdk/keyvault/Azure.Security.KeyVault.Administration/README.md (hunk 2 above)
Comment (AlexGhiondea) [reply sdk/keyvault/Azure.Security.KeyVault.Administration/README.md:160]: Fixed!

Comment (AlexGhiondea) [reply sdk/keyvault/Azure.Security.KeyVault.Administration/README.md]: Maybe? I would be fine if we end up adding a line there as well.

**Comment by AlexGhiondea** [comment]
@heaths I started with KeyVault because I wanted to finalize the text and location for this. Once that's done we can move the text everywhere else.

**Pull request merged**